### DISCUSSION TITLE
It seems not necessary to VIDIOC_QBUF in start_capturing()

### DISCUSSION THREAD

**Discussion by sanbrother** [umlaeute/v4l2loopback#556] [General]
I wrote a sample app based on producer.c
I don't understand why VIDIOC_QBUF in start_capturing(), is it necessary? I removed those lines, and it still works.

The VIDIOC_QBUF breaks poll / select, because it will enqueue N empty buffers.
So, if you capture one image using v4l2-ctl, it will be an empty black one.

```cpp
static void start_capturing(void)
{
	unsigned int i;
	enum v4l2_buf_type type;

	switch (io) {
	case IO_METHOD_WRITE:
		/* Nothing to do. */
		break;

	case IO_METHOD_MMAP:
		for (i = 0; i < n_buffers; ++i) {
			struct v4l2_buffer buf;

			CLEAR(buf);
			buf.type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
			buf.memory = V4L2_MEMORY_MMAP;
			buf.index = i;
			buf.length = buffers[i].length;
			buf.bytesused = buffers[i].bytesused;

			printf("MMAP init qbuf %d/%d (length=%d): %s\n", i,
			       n_buffers, buffers[i].length,
			       snprintf_buffer(strbuf, sizeof(strbuf), &buf));

                        /* WHY should we VIDIOC_QBUF here? */
			if (-1 == xioctl(fd, VIDIOC_QBUF, &buf))
				errno_exit("VIDIOC_QBUF");
		}
		type = V4L2_BUF_TYPE_VIDEO_OUTPUT;
		if (-1 == xioctl(fd, VIDIOC_STREAMON, &type))
			errno_exit("VIDIOC_STREAMON");
		break;
```

**Comment by stephematician** [comment]
By queueing all buffers immediately; the behaviour of the buffers when queueing/dequeuing is that the least-recently queued buffer is always dequeued, and the most recently queued buffer is at the 'head' of the output stream.

**Reply by stephematician** [reply]
> By queueing all buffers immediately; the behaviour of the buffers when queueing/dequeuing is that the least-recently queued buffer is always dequeued, and the most recently queued buffer is at the 'head' of the output stream.

This is a lie.

Actually; queuing and dequeuing of buffers is performed by a _writer_ that has opened the device for output. The _reader_ can queue and dequeue but it has (effectively) no impact on the queue.
